src/model/firebase.js

```diff
@@ -18,32 +19,49 @@ export const getFCMToken = async () => {
 
 export const saveFCMToken = async () => {
   try {
-    const fcmToken = await messaging().getToken();
-    if (fcmToken) {
-      saveLocal('rainbowFcmToken', { data: fcmToken });
+    const permissionStatus = await getPermissionStatus();
+    if (
+      permissionStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      permissionStatus === messaging.AuthorizationStatus.PROVISIONAL
+    ) {
+      const fcmToken = await messaging().getToken();
+      if (fcmToken) {
+        saveLocal('rainbowFcmToken', { data: fcmToken });
+      }
     }
   } catch (error) {
-    logger.log('error getting fcm token - cannot save', error);
+    logger.log('error saving fcm token - cannot save', error);
```

Comment:
just realized that this error msg should actually be about failing to get the fcm token / permission status, not necessarily about saving it